Framework/API/src/WorkspaceGroup.cpp

```diff
@@ -278,21 +279,35 @@ void WorkspaceGroup::workspaceDeleteHandle(
 }
 
 /**
- * Callback when a after-replace notification is received
- * Replaces a member if it was replaced in the ADS.
- * @param notice :: A pointer to a workspace after-replace notificiation object
+ * Callback when a before-replace notification is received
+ * Replaces a member if it was replaced in the ADS and checks
+ * for duplicate members within the group
+ * @param notice :: A pointer to a workspace before-replace notification object
  */
-void WorkspaceGroup::workspaceReplaceHandle(
+void WorkspaceGroup::workspaceBeforeReplaceHandle(
     Mantid::API::WorkspaceBeforeReplaceNotification_ptr notice) {
   std::lock_guard<std::recursive_mutex> _lock(m_mutex);
 
-  const std::string replacedName = notice->objectName();
+  const auto oldObject = notice->oldObject();
+
   for (auto &workspace : m_workspaces) {
```

Comment:
Can we do this whole replace and remove unique in a single loop? It seems a shame to have to loop over the thing twice.